pandas/core/indexes/base.py

```diff
@@ -4948,6 +4952,8 @@ def _get_engine_target(self) -> ArrayLike:
             and not (
                 isinstance(self._values, ArrowExtensionArray)
                 and is_numeric_dtype(self.dtype)
+                # Exclude decimal
+                and self.dtype.kind != "O"
```

Comment:
can we be more specific than this?  e.g. time32 would also go through here

Reply:
`is_numeric_dtype(self.dtype)` should block time32. I could import pyarrow and check `pa.types.is_decimal` if you prefer`

Reply:
makes sense, this is fine